libcst/matchers/_matcher_base.py

```diff
@@ -76,6 +76,18 @@ def __and__(
     def __invert__(self: _BaseMatcherNodeSelfT) -> "_BaseMatcherNodeSelfT":
         return cast(_BaseMatcherNodeSelfT, InverseOf(self))
 
+    def __eq__(self, other: object) -> bool:
+        """
+        Matchers are only treated as equal by identity. This matches the behavior of
+        CPython's AST nodes and LibCST nodes. Comparing matchers by value is useless
+        as they exist only as a way to programatically describe the shape of a tree.
+        """
+        return self is other
+
+    def __hash__(self) -> int:
+        # Equality of nodes is based on identity, so the hash should be too.
+        return id(self)
+
```

Comment:
I think this is the default behavior of `__eq__` and `__hash__` (inherited from `object`) for any Python class that doesn't define them explicitly. So I think it's sufficient for this diff to just turn off the addition of those methods by `@dataclass`; we don't need to define methods that implement the default behavior.

Reply:
Oh, good call. I'll try that.

Reply:
Yup, worked fine. Glad I wrote that test ;)